core/src/main/java/fr/sncf/osrd/api/pathfinding/PathfindingRoutesEndpoint.java

```diff
@@ -102,16 +106,43 @@ public static Pathfinding.Result<SignalingRoute> runPathfinding(
             remainingDistanceEstimator = new RemainingDistanceEstimator(waypoints.get(1));
         }
 
+        Pathfinding.Result<SignalingRoute> pathfinding = computePaths(infra, waypoints, loadingGaugeConstraints,
+                electrificationConstraints, remainingDistanceEstimator);
         // Compute the paths from the entry waypoint to the exit waypoint
-        return new Pathfinding<>(new GraphAdapter<>(infra.getSignalingRouteGraph()))
+        return pathfinding;
+    }
+
+    private static Pathfinding.Result<SignalingRoute> computePaths(SignalingInfra infra,
+             ArrayList<Collection<Pathfinding.EdgeLocation<SignalingRoute>>> waypoints,
+             LoadingGaugeConstraints loadingGaugeConstraints,
+             ElectrificationConstraints electrificationConstraints,
+             RemainingDistanceEstimator remainingDistanceEstimator) {
+
+        var path = new Pathfinding<>(new GraphAdapter<>(infra.getSignalingRouteGraph()))
                 .setEdgeToLength(route -> route.getInfraRoute().getLength())
-                .addBlockedRangeOnEdges(loadingGaugeConstraints)
+                .setRemainingDistanceEstimator(remainingDistanceEstimator);
+        var pathfinding = path.addBlockedRangeOnEdges(loadingGaugeConstraints)
                 .addBlockedRangeOnEdges(electrificationConstraints)
-                .setRemainingDistanceEstimator(remainingDistanceEstimator)
                 .runPathfinding(waypoints);
```

Comment:
We need new instances of Pathfinding for each call. `addBlockedRangesOnEdge` doesn't return a copy but the original instance, it's only used to chain the calls.